### ISSUE TITLE
iocage pkg should use pkg(8) directly

### ISSUE THREAD

**Issue by rwestlund** [iocage/iocage#279]
using HEAD (9a1633aabfe3e84aa1f7861b14741203ff1f7f59).

`iocage pkg` currently passes the given `pkg(8)` command to `jexec(8)` in `ioc_exec.py`.  `stdin` for the child process is set to `PIPE`, meaning that `stdin` is not a tty.  This results in the following message:

```
root@r2d2 /u/h/r/a/s/jails# iocage pkg test2 install nginx
The package management tool is not yet installed on your system.
Please set ASSUME_ALWAYS_YES=yes environment variable to be able to bootstrap in non-interactive (stdin not being a tty)
```
There's a better (and easier) way to accomplish this: `pkg(8)` has a `-j` flag for installing into a jail.  It can be used with the jail's `name` property, or the `jid`:
```
root@r2d2 /u/h/r/a/s/jails# pkg -j ioc-test2 install nginx
root@r2d2 /u/h/r/a/s/jails# pkg -j 19 install nginx
```
`iocage pkg` should call `pkg(8)` directly, and when calling `su.Popen()`, it should set `stdin=None` so that it inherits the fd from the parent.

Aside from allowing interactive usage, it has the additional benefit of not needing to install `pkg(8)` in the jail.

**Comment by skarekrow** [comment]
There was an additional reason I did it this way. I know about the `-j` flag but I can't remember the reasoning on why I d my use it currently. I'll mark this as an enhancement so if I remember I can circle back.
It was something plugin related I believe. So with all that churn it may not even be relevant anymore.

**Comment by rwestlund** [comment]
If only there were some comments in the code explaining why you did it that way... :P 

But seriously, if there is a special use case where you need to run `pkg(8)` inside the jail, just do this:
```
iocage exec test2 pkg install nginx
```

Even for the exec call, I don't know why you create new fds for `stdin` and `stdout` or why you redirect `stderr` to `stdout`.  If you change `su.Popen()` to set `stdin=None, stdout=None, stderr=None`, then everything is inherited from the iocage process, and you can use `iocage exec` with interactive commands just like if you were in the jail.  There's no need for you to grab and buffer all that output.

**Comment by skarekrow** [comment]
I comment plenty :P 

There is, I needed a live stream of output for the plugins, as it just sat there silent as it did it's thing. Atleast that's what my memory tells me.

**Comment by skarekrow** [comment]
I'm going to chalk this up to me overengineering an issue and not hitting my Ballmer Peak! @rwestlund's suggestions work just fine from my tests. I may still be forgetting a usecase, but I don't believe so.

**Comment by skarekrow** [comment]
Fixed in 2f6467b251c52c995fa069a154b7caf54c144e0a, but forgot to put that in the issue...
